tests/unit_test/elife_article_xml/elife_article_xml_pipeline_test.py

```diff
@@ -0,0 +1,324 @@
+
+PROCESSED_FILE_URL_LIST = [PROCESSED_NOT_EMPTY_ARTICLE_URL]
+
+GITHUB_TREE_REPONSE_2 = {
+    'tree': [
+        {
+            'size': 0,
+            'url': EMPTY_ARTICLE_URL
+        },
+        {
+            'size': 10,
+            'url': NOT_EMPTY_ARTICLE_URL_1
+        },
+        {
+            'size': 15,
+            'url': NOT_EMPTY_ARTICLE_URL_2
+        },
+        {
+            'size': 15,
+            'url': PROCESSED_NOT_EMPTY_ARTICLE_URL
+        }
+    ]
+}
+
+GIT_REPO_URL = 'git_repo_url_1'
+
+SOURCE_CONFIG_1 = ElifeArticleXmlSourceConfig(
+    git_repo_url=GIT_REPO_URL,
+    directory_name=MATCHING_DIRECTORY_NAME
+)
+
+TARGET_CONFIG_1 = BigQueryTargetConfig(
+    project_name='project1',
+    dataset_name='dataset1',
+    table_name='table1'
+)
+
+CONFIG_1 = ElifeArticleXmlConfig(
+    source=SOURCE_CONFIG_1,
+    target=TARGET_CONFIG_1
+)
+
+
+XML_FILE_CONTENT_ENCODED = 'TWVyaGFiYSBEdW55YSEgSGVsbG8gV29ybGQh'
+XML_FILE_CONTENT_DECODED = 'Merhaba Dunya! Hello World!'
+
+XML_FILE_JSON = {
+    'content': XML_FILE_CONTENT_ENCODED,
+    'encoding': 'base64'
+}
+
+XML_FILE_URL_LIST = ['xml_file_url_1']
+
+ARTICLE_TYPE_1 = 'article_type_1'
+ARTICLE_ID_1 = 'article_id_1'
+ARTICLE_DOI_1 = 'article_doi_1'
+
+RELATED_ARTICLE_TYPE_1 = 'related_article_type_1'
+RELATED_ARTICLE_DOI_1 = 'related_article_doi_1'
+
+RELATED_ARTICLE_LIST = [
+    {
+        'related-article-type': RELATED_ARTICLE_TYPE_1,
+        '{http://www.w3.org/1999/xlink}href': RELATED_ARTICLE_DOI_1
+    }
+]
+
+XML_FILE_CONTENT_DICT_WITH_RELATED_ARTICLE_SECTION = {
+  'article': {
+    'article-type': ARTICLE_TYPE_1,
+    'front': [
+      {
+        'article-meta': [
+          {
+            'article-id': [
+              {
+                'pub-id-type': 'publisher-id',
+                '_text': ARTICLE_ID_1
+              },
+              {
+                'pub-id-type': 'doi',
+                '_text': ARTICLE_DOI_1
+              }
+            ],
+            'related-article': RELATED_ARTICLE_LIST
+          }
+        ]
+      }
+    ]
+  }
+}
+
+
+class TestGetJsonResponseFromUrl:
+    def test_should_pass_url_to_request_get_function(
+        self,
+        requests_mock: MagicMock
+    ):
+        get_json_response_from_url(url=GIT_REPO_URL)
+        requests_mock.get.assert_called_with(url=GIT_REPO_URL)
+
+
+class TestGetUrlOfXmlFileDirectoryFromRepo:
+    def test_should_return_url_for_matching_path(
+        self,
+        get_json_response_from_url_mock: MagicMock
+    ):
+        get_json_response_from_url_mock.return_value = GITHUB_TREE_REPONSE_1
+        actual_return_value = get_url_of_xml_file_directory_from_repo(SOURCE_CONFIG_1)
+        assert actual_return_value == MATCHING_URL
+
+
+@pytest.mark.usefixtures("get_url_of_xml_file_directory_from_repo_mock")
+class TestIterXmlFileUrlFromGitDirectory:
+    def test_should_return_url_list_only_for_not_empty_and_non_processed_files(
```

Comment:
we then also should have a test, that it is skipping zero length files, and already processed files.

Reply:
testing it too